ThirdPartyAdapters/unity/unity/src/main/java/com/google/ads/mediation/unity/UnityAdapter.java

```diff
@@ -263,6 +269,9 @@ public void onUnityAdsShowFailure(String placementId, UnityAdsShowError error, S
       // Unity Ads ad failed to show.
       String sdkError = createSDKShowError(error, message);
       Log.w(TAG, "Unity Ads returned an error: " + sdkError);
+      if (mMediationInterstitialListener != null && error == UnityAdsShowError.NOT_READY) {
+        mMediationInterstitialListener.onAdClosed(UnityAdapter.this);
+      }
```

Comment:
Why are we only closing for NOT_READY.  what about other error codes?

Reply:
Its to keep it as it was before when the isReady returned false, didn't ask myself the question.
But good point. We never called onAdClosed before when errors happened, perhaps we should now for every errors? (though technically, the Ad never "opened" as show failed. hmmm)

Reply:
ok close for this pr, since its maintaining status quo.  we should question this for the 3.7.1 refactor work.